ttnn/cpp/ttnn/operations/data_movement/reshape_view/reshape.cpp

```diff
@@ -167,6 +167,12 @@ ttnn::Tensor ReshapeViewOperation::invoke(const ttnn::Tensor& tensor, const ttnn
     if (tensor_shape == shape) {
         return tensor;
     }
+
+    // Just edit shape if shape has a 0 dimension
+    if tensor.volume() == 0 {
+        return tensor.reshape(shape);
+    }
+
```

Comment:
We should also check shape.volume which should be 0 if and only if tensor.volume is 0. Otherwise we can hit errors down the road

Reply:
Ok I'll check

Reply:
Could you also add:
TT_ASSERT(shape.volume() != 0 && "tensor's volume is not 0, but shape's volume is 0");
Right after the if just to cover us in both directions?

Reply:
okok I added, thank you